modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/management_bar/components/filters/CheckboxesFilter.js

```diff
@@ -13,100 +13,179 @@
  */
 
 import ClayButton from '@clayui/button';
-import {ClayCheckbox} from '@clayui/form';
+import ClayDropDown from '@clayui/drop-down';
+import {ClayCheckbox, ClayToggle} from '@clayui/form';
 import PropTypes from 'prop-types';
-import React, {useState} from 'react';
+import React, {useEffect, useState} from 'react';
 
-function getOdataString(value, key) {
-	if (!value || !value.length) {
+import {isValuesArrayChanged} from '../../../utilities/index';
+
+export const formatValue = (value, items, exclude) => {
+	const formattedValue = value
+		? value
+				.map((v) => {
+					return items.reduce(
+						(found, item) =>
+							found || (item.value === v ? item.label : null),
+						null
+					);
+				})
+				.join(', ')
+		: '';
+
+	return (
+		(exclude ? `(${Liferay.Language.get('exclude')}) ` : '') +
+		formattedValue
+	);
+};
+
+function getOdataString(values, key, exclude = false) {
+	if (!values || !values.length) {
 		return null;
 	}
 
-	return `${key}/any(x:${value
+	return `${key}/any(x:${values
 		.map(
 			(v) =>
-				`(x eq ${
-					typeof v.value === 'string' ? `'${v.value}'` : v.value
+				`(x ${exclude ? 'ne' : 'eq'} ${
+					typeof v === 'string' ? `'${v}'` : v
 				})`
 		)
-		.join(' or ')})`;
+		.join(exclude ? ' and ' : ' or ')})`;
 }
-
 function CheckboxesFilter({actions, id, items, value: valueProp}) {
-	const [value, setValue] = useState(valueProp);
+	const [itemsValues, setItemsValues] = useState(
+		(valueProp && valueProp.itemsValues) || []
+	);
+	const [exclude, setExclude] = useState(
+		valueProp ? valueProp.exclude : false
+	);
 
-	function selectCheckbox(itemValue) {
-		if (!value) {
-			setValue([itemValue]);
-		}
-		else if (!value.includes(itemValue)) {
-			setValue(value.concat(itemValue));
-		}
-		else if (value.length === 1) {
-			setValue(undefined);
-		}
-		else {
-			setValue(value.filter((v) => v !== itemValue));
+	function selectCheckbox(selected) {
+		if (itemsValues.includes(selected)) {
+			return setItemsValues(itemsValues.filter((v) => v !== selected));
```

Comment:
We generally try to avoid abbreviations for variable names like (el -> element or v -> value)